### DISCUSSION TITLE
How to connect to web3modal?

### DISCUSSION THREAD

**Discussion by smartcoder0305** [wevm/wagmi#2373] [Q&A]
```
const chains = [arbitrum, mainnet, polygon];
const projectId = process.env.NEXT_PUBLIC_YOUR_PROJECT_ID as string;

const { provider } = configureChains(chains, [publicProvider()]);
const wagmiClient = createClient({
  autoConnect: true,
  provider,
});
const ethereumClient = new EthereumClient(wagmiClient, chains);
```

The code above is the connecting part between wagmi and web3modal.
The code below is the component.

```
...
      <Web3Modal
        projectId={projectId}
        explorerExcludedWalletIds={"ALL"}
        explorerRecommendedWalletIds={[
          process.env.NEXT_PUBLIC_METAMASK_WALLET_ID as string,
          process.env.NEXT_PUBLIC_COINBASE_WALLET_ID as string,
        ]}
        ethereumClient={ethereumClient}
      />
...
```

And it's throwing this error.
Error: WalletConnectConnector or WalletConnectLegacyConnector is required

I think it's because the wagmi library is updated 3 days ago.
Please let me know how to fix this.
Thanks in advance.

**Comment by SaadNiaziDev** [comment]
``
const wagmiClient = createClient({
  autoConnect: true,
  connector:[  any connector u want to use]
  provider,
});
``
need to fix code litle bit
